addons/mail/static/src/owl/components/composer/composer.scss

```diff
@@ -23,6 +23,47 @@
             padding-top: 20px;
         }
     }
+
+    &:not(.o-is-compact) {
```

Comment:
prefer using modifier on each item, so that there are almost no css specificity involved